[Fix] Broken behavior : `make generate` command fails when only using plans in map config notation

## Impacted version

[0.2.21](https://github.com/Orange-OpenSource/AWSTerraformStarterKit/releases/tag/v0.2.21), [0.2.22](https://github.com/Orange-OpenSource/AWSTerraformStarterKit/releases/tag/v0.2.22)

## Steps to reproduce

Starting from a fresh or already existing project run the following tasks
1. Run `make down`
2. Remove your `.env` file with `rm .env`
3. Remove the current `jinja2docker` image with the following command: `docker rmi -f $(docker image ls -a | grep jinja2docker | awk '{print $3}')`
4. Make sure you are using one of the `0.2.21` or `0.2.22` version. Check the `.STARTER_KIT_CURRENT_VERSION` file content or simply override it with `echo "0.2.22" | tee .STARTER_KIT_CURRENT_VERSION`
5. Run `./get-starter-kit.sh`
6. Update your `configure.yaml` `plans` list with the following

```yaml
########################################################################################################################
# PLANS
########################################################################################################################

...

# List the folder for each Terraform Plan to install, respect the order of installation
# Replace the default demo plan by your own plan name and configuration
plans:
- name: example/complete
  override_init_parameters: ""
  override_var_parameters: ""

...
```

5. Run the `make generate` command

You should get the following error

```txt
Traceback (most recent call last):
  File "/scripts/entities/render.py", line 89, in <module>
    main()
  File "/scripts/entities/render.py", line 85, in main
    sys.stdout.write(Render(sys.argv[1], sys.argv[2]).rend_template())
  File "/scripts/entities/render.py", line 56, in rend_template
    var_data['plans'][idx] = var_data['plans'][idx] if var_data['plans'][idx].startswith('name: ') else {
AttributeError: 'dict' object has no attribute 'startswith'
make[1]: *** [Makefile:279 : init] Erreur 1
```

Caused by the not type safe code in the `automation/jinja2/render.py` file:

```python
 var_data['plans'][idx] = var_data['plans'][idx] if var_data['plans'][idx].startswith('name: ') else {
```

--- a/automation/jinja2/render.py
+++ b/automation/jinja2/render.py
@@ -53,7 +53,7 @@ class Render:
             var_data = yaml.full_load(f)
         # compute execution plan
         for idx in range(len(var_data['plans'])):
-            var_data['plans'][idx] = var_data['plans'][idx] if var_data['plans'][idx].startswith('name: ') else {
+            var_data['plans'][idx] = var_data['plans'][idx] if not isinstance(var_data['plans'][idx],str) else {
                 'name': var_data['plans'][idx]} # this is for backward compatiblity
         exec_plan = compute_deps.build_exec_plan(plans=var_data['plans'])
         var_data['exec_plan'] = exec_plan

